pandas/core/dtypes/cast.py

```diff
@@ -723,19 +718,37 @@ def astype_nansafe(arr, dtype, copy=True, skipna=False):
 
     elif is_object_dtype(arr):
 
-        # work around NumPy brokenness, #1987
-        if np.issubdtype(dtype.type, np.integer):
-            return lib.astype_intsafe(arr.ravel(), dtype).reshape(arr.shape)
-
         # if we have a datetime/timedelta array of objects
         # then coerce to a proper dtype and recall astype_nansafe
 
+        if (is_timedelta64_dtype(dtype)
+                or np.issubdtype(dtype.type, np.integer)):
+            # TODO: this is an old numpy compat branch that is not necessary
+            # anymore for its original purpose (unsafe casting from object to
+            # int, see GH 1987).
+            # Currently, timedelta dtypes get routed through here; whereas
+            # uncommenting them would re-call (see below)
```

Comment:
these comments are way too long. what exactly are you trying to do here?  let's leave this entire casting as is for now w/o any changes. this needs a separate followup.

Reply:
I'm trying to leave some breadcrumbs as I spent 2-3 hours trying to investigate this.

I've shortened the comment now.

Reply:
i would rather you not and just revert back to what it was. if you want to make an issue ok.